### PR TITLE
vispy: init at 0.6.4

### PR DESCRIPTION
**Pull request by goertzenator**
###### Motivation for this change
Add python module for `vispy` (scientific plotting) and `freetype-py` (needed by `vispy`).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/freetype-py/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/freetype-py/default.nix]: nativeBuildInputs

File: pkgs/development/python-modules/vispy/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/vispy/default.nix]: aside from pyqt4 the others likely belong in `nativeBuildInputs`. `setuptools` should not be here, unless it's needed at runtime in which case it should be in `propagatedBuildInputs`

File: pkgs/development/python-modules/freetype-py/default.nix
Comment (goertzenator) [reply pkgs/development/python-modules/freetype-py/default.nix]: updated as requested

File: pkgs/development/python-modules/vispy/default.nix
Comment (goertzenator) [reply pkgs/development/python-modules/vispy/default.nix]: updated as requested, `setuptools` and `pyqt4` removed (`pyqt4` was part of a failed effort to make tests work.)

File: pkgs/development/python-modules/vispy/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/vispy/default.nix]:
```suggestion
    homepage = "http://vispy.org/index.html";
```

File: pkgs/development/python-modules/freetype-py/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/freetype-py/default.nix]:
```suggestion
    homepage = "https://github.com/rougier/freetype-py";
```

**Review by jonringer — COMMENTED** [review]
mostly LGTM.

Can you add yourself as a maintainer of the packages?
Also, can you try to add unit tests (if possible). Don't need 100% of the test suite, they just help when bumping dependencies and ensure something doesn't break.

**Comment by aanderse** [comment]
(triage) ping @goertzenator

**Comment by goertzenator** [comment]
Thanks for reviewing, changes made as requested.

I ended up not using this package for my project so my interest in learning how to add unit tests is low.  I did run `vispy` tests manually and there were *lots* of failures, but it did work for my brief experiment.  I suspect the failures are more about the maturity of the project than interaction with nixpkgs.

**Comment by mjlbach** [comment]
I'm interested in getting this merged. Is this just blocked by tests?

**Comment by goertzenator** [comment]
I don't know what is blocking it.  I think it might be helpful if you tried this PR on your system and reported if it produced a working vispy or not.

**Comment by FRidh** [comment]
Please avoid merging Nixpkgs branches and instead rebase.

**Comment by goertzenator** [comment]
Apologies, I just hit buttons in the github UI to resolve a minor conflict.  I didn't see any choices about how to manage it.  I'll tidy this up on the command line.

That said, maybe some github settings need to be tweaked if those buttons are doing the wrong thing?

**Comment by goertzenator** [comment]
All cleaned up.  I looked at the settings on one of my repos and unfortunately did not see any controls for the "Resolve Conflicts" button.

**Comment by mjlbach** [comment]
Currently having a slight issue...

## shell.nix ##

```
{ pkgs ? import <nixpkgs> {} }:
  pkgs.mkShell {
    buildInputs = [
      pkgs.python3Packages.vispy
      pkgs.python3Packages.pyqt5
  ];
}
```
## error msg ##
```
WARNING: Could not find the Qt platform plugin "xcb" in ""
WARNING: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: wayland-org.kde.kwin.qpa.

Aborted (core dumped)
```

This is usually fixed by adding a wrapQtAppsHook https://nixos.wiki/wiki/Qt, I know vispy uses pyqt5 but not sure what's happening here.

**Comment by goertzenator** [comment]
Looking back at my `shell.nix` I note that I had `pyqt4` instead of `pyqt5`

**Comment by mjlbach** [comment]
@goertzenator Thanks! That seems to do it... strange, wispy should be compatible with PyQt5

**Comment by mjlbach** [comment]
Nevermind, I just tried again this morning and it seems to be working with PyQt5, i tried a dozen examples and they all seem to work. Thanks for the contribution!

File: pkgs/development/python-modules/vispy/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/vispy/default.nix:31]:
```suggestion
  doCheck = false;  # otherwise runs OSX code on linux.
  pythonImportsCheck = [ "vispy" ];
```

File: pkgs/development/python-modules/freetype-py/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/freetype-py/default.nix:22]:
```suggestion

  pythonImportsCheck =  [ "freetype" ];
```

**Review by jonringer — COMMENTED** [review]
I would prefer unit tests, but this should suffice

**Review by prusnak — APPROVED** [review]
Tested ACK

**Comment by prusnak** [comment]
@jonringer every suggested change has been applied, can we have this merged?

**Pull request merged**
